File: JavaScript/07_loops.js
#!/usr/bin/env node

function loops() {
    console.log("=== LOOPS ===");

    // For loop
    console.log("For loop:");
    for (let i = 0; i < 5; i++) {
        process.stdout.write(`  For: ${i} `);
    }
    console.log();

    // For-of (range-based)
    const numbers = [10, 20, 30];
    console.log("For-of:");
    for (const num of numbers) {
        process.stdout.write(`  ${num} `);
    }
    console.log();

    // While
    console.log("While loop:");
    let count = 0;
    while (count < 3) {
        console.log(`  While: ${count}`);
        count++;
    }

    // Array methods (very JS-idiomatic)
    console.log("Array map:");
    const squares = [1, 2, 3, 4, 5].map(x => x * x);
    console.log(squares);
}

function main() {
    loops();
}

main();

File: Python/07_loops.py
#!/usr/bin/env python3

def loops():
    """04_loops.py - for, while, list comprehensions"""
    # For loop
    print("For loop:")
    for i in range(5):
        print(f"  For: {i}", end=" ")
    print()

    # For-each style
    numbers = [10, 20, 30]
    print("For-each:")
    for num in numbers:
        print(f"  {num}", end=" ")
    print()

    # While loop
    print("While loop:")
    count = 0
    while count < 3:
        print(f"  While: {count}")
        count += 1

    # List comprehension (Pythonic)
    squares = [x * x for x in range(6)]
    print(f"List comprehension (squares): {squares}")

def main():
    loops()

if __name__ == '__main__':
    main()
